### PR TITLE
Add exceptions declaring guidelines

### PR DESCRIPTION
**Pull request by adaszko**
I'm thinking we could get rid of Jekyll altogether and make a simple markdown file out of it.  GitHub rendering of Markdown should be enough for our needs and will make it easier to contribute to.

I also removed the changelog since there's Git history.

### CODE REVIEW

**Review by marzetas — COMMENTED** [review]
+1 for the new section. 

I'd would keep the example on how to catch multiple exceptions in a piece of code, I remember it landed here because people would do that in different ways, and I think the one in the old docs was the better one.

File: python.md [hunk 1]
```diff
@@ -31,35 +31,77 @@ d = {
...
+
+**Name** your exceptions [in accordance with
+PEP8](https://www.python.org/dev/peps/pep-0008/#exception-names), i.e. use
+suffixes like `Error` or `Warning` where appropriate.
+
+**Avoid** passing human-readable messages to exception constructors, prefer
+passing problematic variable values instead and naming the exception class
+descriptively.  Rationale: Exceptions are entities belonging to the realm of
+the code, not to that of the users.  Users aren't being shown raw exceptions,
+they're being show error messages which are an interpretation of cought
```
Comment (ldoran) [inline python.md]: 2 typos: s/being show/being shown/ and s/cought/caught/

File: python.md [hunk 2]
```diff
@@ -31,35 +31,77 @@ d = {
...
+exceptions.   There are considerations like localization and formatting of the
+message that need to be taken into account before showing a message to the
+user.  Trying to address these consideration at the point in code at which the
```
Comment (ldoran) [inline python.md]: Feel free to ignore this suggestion: I'd write "Trying to address such considerations"

**Review by ldoran — COMMENTED** [review]
Nice one.

+1 on Raul's suggestion about multiple exceptions.

File: python.md (hunk 2 above)
Comment (adaszko) [reply python.md]: That's too subtle to a non-native 😉 Corrected.

**Review by ldoran — APPROVED** [review]

File: python.md [hunk 3]
```diff
@@ -31,10 +31,74 @@ d = {
...
+Exception classes you declare **should inherit** (either directly or indirectly)
+from the `Exception` class.  If there's more that can be told about the nature
+of the exceptional situation, pick a more specific exception class to inherit
+from, from the hierarchy of [predefined exceptions](https://docs.python.org/2/library/exceptions.html#exception-hierarchy).
+
+The last parameter of `__init__()` method **should be `*args`** to allow for
+passing arbitrary number of arguments, which in turn are passed on to
+`super(...).__init__(..., *args)`.  That way, the Liskov Substitution
+Principle isn't violated between your exception class and the `Exception`
+class.
```
Comment (adaszko) [inline python.md:66]: @ldoran: Sorry for pushing after the PR was accepted but I thought another section about `*args*` was needed.

Comment (ldoran) [reply python.md:66]: No worries, I looked at the update and I shippit.

**Review by ldoran — APPROVED** [review]

**Comment by adaszko** [comment]
@marzetas 

> I'd would keep the example on how to catch multiple exceptions in a piece of code, I remember it landed here because people would do that in different ways, and I think the one in the old docs was the better one.

If you're referring to this section:

> -* When writing controllers which handle a lot of Exceptions, try to encapsulate the logic within one `try..except`

It was simply moved, not deleted.

**Review by marzetas — APPROVED** [review]

**Pull request merged**